components/scroller/scroller.vue

```diff
@@ -0,0 +1,163 @@
+<template>
+  <component
+    :is="dynamic ? 'dynamic-scroller' : 'core-scroller'"
+    ref="scroller"
+    data-qa="dt-scroller"
+    :items="items"
+    :item-size="itemSize"
+    :min-item-size="minItemSize"
+    :direction="direction"
+    key-field="id"
+    :list-tag="listTag"
+    :item-tag="itemTag"
+    v-bind="$attrs"
+    :style="computedStyle"
+  >
+    <template
+      #default="{ item, index, active }"
+    >
+      <slot
+        v-bind="{
+          item: item,
+          index,
+          active,
+        }"
+      />
+    </template>
+  </component>
+</template>
+
+<script>
+import CoreScroller from './modules/core-scroller.vue';
+import DynamicScroller from './modules/dynamic-scroller.vue';
+
+export default {
+  name: 'DtScroller',
+
+  components: {
+    CoreScroller,
+    DynamicScroller,
+  },
+
+  provide () {
+    // use function syntax so that we can access `this`
+    return {
+      emit: this.$emit,
+    };
+  },
+
+  inheritAttrs: false,
+
+  props: {
+    /**
+      * The items to render.
+      * If the items are simple arrays, the index will be used as the key.
+      * If the items are objects, the keyField will be used as the key.
+     */
+    items: {
+      type: Array,
+      required: true,
+    },
+
+    /**
+      * The width of the scroller.
+      * Can be a number (in pixels) or a string (in CSS units).
+     */
+    scrollerWidth: {
+      type: [String, Number],
+      default: '100%',
+    },
+
+    /**
+      * The height of the scroller.
+      * Can be a number (in pixels) or a string (in CSS units).
+     */
+    scrollerHeight: {
+      type: [String, Number],
+      default: '100%',
+    },
+
+    /**
+      * Indicates if the items are dynamic.
+      * If true, the items will be wrapped in a DtScrollerItem component.
+      * This is required for dynamic items to be able to react to changes in their size.
+     */
+    dynamic: {
+      type: Boolean,
+      default: false,
+    },
+
+    /**
+      * The key field to use for the items.
+      * If the items are objects, the scroller needs to be able to identify them.
+      * By default it will look for an id field on the items.
+      * This can be configured with this prop if you are using another field name.
+     */
+    keyField: {
+      type: String,
+      default: 'id',
+    },
+
+    /**
+      * The direction of the scroller.
+      * Can be either 'vertical' or 'horizontal'.
+     */
+    direction: {
+      type: String,
+      default: 'vertical',
+      validator: (value) => ['vertical', 'horizontal'].includes(value),
+    },
```

Comment:
Added @values true, false on `dynamic` prop
Added @values vertical, horizontal on `direction` prop

Reply:
Should we additionally add 
@type
@required
@default 
@example

Example on `Items` prop:

@type {Array}
@required
@default []
@example items: [ 'item1', 'item2', 'item3' ]
@example items: [ { id: 1, name: 'item1' }, { id: 2, name: 'item2' }, { id: 3, name: 'item3' } ]

Reply:
I believe we don't need type/required/default because we get it directly from the vue prop. Example is kinda nice though, I'll leave it up to you if you think it's necessary.